OpenRA.Mods.Cnc/Traits/Mine.cs

```diff
@@ -59,11 +59,13 @@ bool ICrushable.CrushableBy(Actor self, Actor crusher, BitSet<CrushClass> crushC
 			return info.CrushClasses.Overlaps(crushClasses);
 		}
 
-		bool ICrushable.TryCalculatePlayerBlocking(Actor self, BitSet<CrushClass> crushClasses, out LongBitSet<PlayerBitMask> blocking)
+		LongBitSet<PlayerBitMask> ICrushable.CrushableBy(Actor self, BitSet<CrushClass> crushClasses)
 		{
-			// Fall back to the slow path
-			blocking = default(LongBitSet<PlayerBitMask>);
-			return false;
+			var blocking = info.BlockFriendly ? self.Owner.AllyMask : default(LongBitSet<PlayerBitMask>);
+
+			blocking = blocking.Union(info.CrushClasses.Overlaps(crushClasses) ? default(LongBitSet<PlayerBitMask>) : self.World.AllPlayerMask);
```

Comment:
Am I right in assuming this can be simplified down to
```csharp
if (!info.CrushClasses.Overlaps(crushClasses))
	return self.World.AllPlayerMask;

// Friendly units should move around!
return info.BlockFriendly ? self.Owner.AllyMask : default(LongBitSet<PlayerBitMask>);
```
?

Reply:
What do you think about introducing a `self.World.NoPlayersMask` (and renaming `AllPlayersMask`, `AlliedPlayersMask`, `EnemyPlayersMask` for consistency)?

IMO this would make code like this a lot more intuitive.